server.js

```diff
@@ -0,0 +1,309 @@
+  }
+
+  if (!req.headers.authorization) {
+    res.writeHead(401, { 'WWW-Authenticate': 'Basic realm="Secure Area"' });
+    res.end('<html><body>Not Authorized</body></html>');
+  } else {
+    const encodedString = req.headers.authorization.slice(6);
+    const base64Buffer = new Buffer(encodedString, 'base64');
+    const decodedString = base64Buffer.toString();
+    const loginArray = decodedString.split(':');
+    const username = loginArray[0];
+    const password = loginArray[1];
+
+    if (!(Users[username] === password)) {
+      res.writeHead(401, { 'WWW-Authenticate': 'Basic realm="Secure Area"' });
+      res.end('<html><body>Not Authorized</body></html>');
+    } else {
+      if (req.method === 'POST') {
+        if (req.url !== '/elements') {
+          res.writeHead(404);
+          return res.end('Could not post to file path: ' + req.url);
+        } else {
+          const requiredFields = [
+            'elementName',
+            'elementSymbol',
+            'elementAtomicNumber',
+            'elementDescription'
+          ];
+          let hasRequiredFields = true;
+          let requestBody = '';
+          let parsedRequestBody = '';
+          let htmlFileText = '';
+          let indexHTMLText = '';
+
+          req.on('data', chunk => {
+            requestBody += chunk;
+          });
+
+          req.on('end', chunk => {
+            if (chunk) {
+              requestBody += chunk;
+            }
+
+            parsedRequestBody = querystring.parse(requestBody);
+
+            requiredFields.forEach(field => {
+              if (!Object.keys(parsedRequestBody).includes(field)) {
+                hasRequiredFields = false;
+              }
+            });
+
+            if (!hasRequiredFields) {
+              res.writeHead(406);
+              res.end('request does not provide required fields');
+            } else {
+              htmlFileText = `<!DOCTYPE html>
+        <html lang="en">
+          <head>
+            <meta charset="UTF-8" />
+            <title>The Elements - Helium</title>
+            <link rel="stylesheet" href="/css/styles.css" />
+          </head>
+          <body>
+            <h1>${parsedRequestBody.elementName}</h1>
+            <h2>${parsedRequestBody.elementSymbol}</h2>
+            <h3>Atomic number ${parsedRequestBody.elementAtomicNumber}</h3>
+            <p>
+              ${parsedRequestBody.elementDescription}
+            </p>
+            <p><a href="/">back</a></p>
+          </body>
+        </html>
+        `;
+
+              fs.writeFile(
+                `./public/${parsedRequestBody.elementName}.html`,
+                htmlFileText,
+                (err, data) => {
+                  if (err) {
+                    res.writeHead(500, { 'Content-Type': 'application/JSON' });
+                    res.end('{ "success": false }');
+                  }
+                }
+              );
+
+              fs.readFile('./public/index.html', 'utf8', (err, data) => {
```

Comment:
This readFile should in the writeFile above. If there was an error writing the file, you don't want to update the index with a file that's not there.